ldappool/__init__.py

```
@@ -0,0 +1,449 @@
+
+
+class ConnectionPool(object):
+    # OPTIMIZATION: Use __slots__
+    __slots__ = ("uri", "binddn", "bindpw", "params", "max", "_lock", "_pool", "_context_conn")
+
+    def __init__(
+        self,
+        uri: LDAPUrl = None,  # Avoid mutable default, handled below or in field def
+        binddn: str = "",
+        bindpw: str = "",
+        params: dict = {},  # type: ignore
```

Ruff (0.14.8)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)